### ISSUE TITLE
Improve discoverability of drag-and-drop feature

### ISSUE THREAD

**Issue by shihern** [beemdevelopment/Aegis#492]
Hi there! I have been using Aegis for a while and love the features it provides.

Aegis currently supports a drag and drop feature for reordering items with the following procedure:
1. Long press an item, then release
2. Long press the item again and hold
3. Drag the item to the desired location

This is unintuitive and hard for users to discover. It seems I'm not the only one too: #482 #322 #357

In most other apps, drag and drop can be done by simply long-pressing the item and dragging. There is no need to release before selecting the item again. Alternatively, many apps add a drag handle which the user can use to move items around. It would be great if you could implement something similar to improve usability.

Based on what I've seen of your code, you should be able to implement this using the [startDrag function in ItemTouchHelper](https://developer.android.com/reference/androidx/recyclerview/widget/ItemTouchHelper#startDrag(androidx.recyclerview.widget.RecyclerView.ViewHolder)). If you want, I can try my hand at implementing this myself (though it may take a while as I'm not familiar with your codebase).

I hope you'll take my suggestion into account. Either way, thanks so much for making this great app! 

Example of drag handles in Spotify:
![photo_2020-06-26_05-22-42](https://user-images.githubusercontent.com/5990763/85728862-d2e2c900-b6e7-11ea-9540-776860b2cc7b.jpg)

**Comment by michaelschattgen** [comment]
Thanks for your suggestions! We agree that it's very intuitive, we just haven't found a better way.

>In most other apps, drag and drop can be done by simply long-pressing the item and dragging

What do you think of the way the new Google Authenticator implemented this? It adds a drag handle which makes the user aware of the drag and drop feature. This combined with immediately triggering the drag-and-drop after the next click (so step 2 can be replaced with step 3), would that work for you?

>many apps add a drag handle which the user can use to move items around

I'm not a fan of this, that means we need to add some sort of edit mode which allows the users to drag and drop every entry.

**Comment by shihern** [comment]
Thanks for the reply!

Yes I think the Google authenticator method would be a great option. This would make it much more intuitive for new users.

**Comment by michaelschattgen** [comment]
Great! Is this something you would like to work on?

**Comment by shihern** [comment]
Ok sure I'll see what I can do. I'll comment here if I need any help. Thanks!

**Comment by michaelschattgen** [comment]
Awesome, we're more than happy to help.

**Comment by shihern** [comment]
Hi, I am having some difficulties implementing this feature.

Here is what currently happens in the following 2 scenarios:

Assume there are 3 entries: x, y, and z

## Scenario A
|Step|User Action|Dragging Behavior in current release| 
|---|---|---|
|  1 |  Press and hold entry x to select it  |  Dragging enabled for entry x |
|  2 |  Select additional entries y and z | Dragging disabled entirely |
|  3 |  Deselect entries y and z, leaving only the original selection (entry x) selected | Dragging re-enabled for entry x |

## Scenario B
|Step|User Action|Dragging Behavior in current release| 
|---|---|---|
|  1 |  Press and hold entry x to select it  |  Dragging enabled for entry x |
|  2 |  Select additional entries y and z | Dragging disabled entirely |
|  3 |  Deselect entries x and y, leaving only entry z selected | Dragging still disabled |

How should I implement the dragging behavior?
Options:
1. Keep the current behavior
2. Enable dragging on step 3 for the remaining item, regardless of whether it is the original selection
3. Disable dragging entirely after step 2 until the user deselects ALL entries (i.e. exits the edit mode)

Also, how much leeway do I have in changing the behavior of existing functions? It will be very helpful instead of having to tack on more functions to make the changes I want.

Thanks for the help!

**Comment by alexbakker** [comment]
The behavior you describe in Scenario B is not intended behavior. It's a bug. If only one entry is selected, it should be draggable, regardless of whether it was the first to be selected. If you'd like, you can take a crack at fixing that as part of your change. We can also create a new issue and solve that separately. I'd go with option 2 of the ones you listed.

> Note: if dragging is reenabled on step 3, the drag handle will have to reappear too. I'm not sure if this is good UX as the drag handle will appear on an entry the user did not interact with (e.g. user taps z to deselect it, drag handle suddenly appears on x)

I think that's fine, as it ensures consistency.

> Also, how much leeway do I have in changing the behavior of existing functions? It will be very helpful instead of having to tack on more functions to make the changes I want.

Any modifications to the code that are in-scope of your change are fine.
